src/analysis/retail/warlock/demonology/modules/talents/DiabolicRitualEmpowers.tsx

```diff
@@ -0,0 +1,58 @@
+import { formatThousands } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import TALENTS from 'common/TALENTS/warlock';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { DamageEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+
+class DiabolicRitualEmpowers extends Analyzer {
+  ruination_damage = 0;
+  infernal_bolt_damage = 0;
+  // TODO : calculate Cloven souls damage
+  // cloven souls is a debuff on enemy, all damage warlock do is increased by 5%
+  // calculate damage done when debuff is up and get 5% from it
+  // see Arcane Mage - Touch of the magi
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS.DIABOLIC_RITUAL_TALENT);
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.RUINATION_DAMAGE),
+      this.handleFelseekerDamage,
+    );
+    this.addEventListener(
```

Comment:
Similiar thing here where you can condense the event listener to 1 by adding an array of relevant spells to the filter